test/jdk/tools/jpackage/helpers/jdk/jpackage/test/TestMethodSupplier.java

```diff
@@ -0,0 +1,512 @@
+        }
+
+        return new Parameter[0];
+    }
+
+    private static ParameterSupplier[] getMethodParameterSuppliers(Method method) {
+        if (method.isAnnotationPresent(ParameterSupplierGroup.class)) {
+            return ((ParameterSupplierGroup) method.getAnnotation(ParameterSupplierGroup.class)).value();
+        }
+
+        if (method.isAnnotationPresent(ParameterSupplier.class)) {
+            return new ParameterSupplier[]{(ParameterSupplier) method.getAnnotation(
+                ParameterSupplier.class)};
+        }
+
+        return new ParameterSupplier[0];
+    }
+
+    private static Stream<Method> filterParameterSuppliers(Class<?> type) {
+        return Stream.of(type.getMethods())
+                .filter(m -> m.getParameterCount() == 0)
+                .filter(m -> (m.getModifiers() & Modifier.STATIC) != 0)
+                .sorted(Comparator.comparing(Method::getName));
+    }
+
+    private static Stream<Object[]> createArgs(Method ... parameterSuppliers) throws
+            IllegalAccessException, InvocationTargetException {
+        List<Object[]> args = new ArrayList<>();
+        for (var parameterSupplier : parameterSuppliers) {
+            args.addAll((Collection) parameterSupplier.invoke(null));
+        }
+        return args.stream();
+    }
+
+    private static Object fromString(String value, Class toType) {
+        if (toType.isEnum()) {
+            return Enum.valueOf(toType, value);
+        }
+        Function<String, Object> converter = FROM_STRING.get(toType);
+        if (converter == null) {
+            throw new RuntimeException(String.format(
+                    "Failed to find a conversion of [%s] string to %s type",
+                    value, toType.getName()));
+        }
+        return converter.apply(value);
+    }
+
+    private static void trace(String msg) {
+        if (TKit.VERBOSE_TEST_SETUP) {
+            TKit.log(msg);
+        }
+    }
+
+    static class InvalidAnnotationException extends Exception {
+        InvalidAnnotationException(String msg) {
+            super(msg);
+        }
+    }
+
+    private static class Verifier {
+        static boolean isTestClass(Class<?> type) {
+            for (var method : type.getDeclaredMethods()) {
+                method.setAccessible(true);
+                if (isParameterized(method) || isTest(method)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        static void verifyTestClass(Class<?> type) throws InvalidAnnotationException {
+            boolean withTestAnnotations = false;
+            for (var method : type.getDeclaredMethods()) {
+                method.setAccessible(true);
+                if (withTestAnnotations) {
+                } else if (isParameterized(method) || isTest(method)) {
+                    withTestAnnotations = true;
+                }
+                verifyAnnotationsCorrect(method);
```

Comment:
This code blocks looks confusing. Why we need `method.setAccessible(true)` for all methods? Also, `if` statement looks confusing with empty `if (withTestAnnotations).`

Reply:
Good point about `method.setAccessible(true)`!
Looks like it is needed when `Method::invoke()` is used, but not when querying annotations. Removed redundant `method.setAccessible(true)` calls.